### PR TITLE
Rename `suspend_display()` to `hide()`

### PR DESCRIPTION
**Pull request by wch**
This PR renames `suspend_display` to `hide`. The old name feels very advanced, and it also sounds imperative and stateful, and it refers to a state that most users don't really need to understand and could be confusing. For example, a user might thing, "OK, so we were displaying before, but now we're suspending it... and then how do I resume it?"

Here are some examples of usage:

```py
@suspend_display
@render.text()
def txt():
    return "Text here"


with suspend_display():
    @render.text()
    def txt2():
        return "Text here"


ui.output_text("txt", inline=True)
ui.output_text("txt2")
```

With `hide()`, it's a bit less confusing:

```py
@hide
@render.text()
def txt():
    return "Text here"


with hide():
    @render.text()
    def txt2():
        return "Text here"


ui.output_text("txt", inline=True)
ui.output_text("txt2")
```

One possibly confusing thing about using `hide` or `hidden` is that it might suggest to the user that the `render`ed content won't be shown at all, when in fact it will be shown where the corresponding `output_*()` function is used.

### CODE REVIEW

**Comment by gadenbuie** [comment]
> One possibly confusing thing about using `hide` or `hidden` is that it might suggest to the user that the `render`ed content won't be shown at all, when in fact it will be shown where the corresponding `output_*()` function is used.

To avoid this connotation, I was thinking of proposing `@hold` for the decorator. It conveys the sense that Shiny Express is holding back the UI, but it's still available to be used.

**Comment by gadenbuie** [comment]
Maybe we could hang this decorator and `@display_body` together with something like `@display_hide`, `@display_hold` or `@display_suspend`. While I have argued against `@suspend_display`, I think my concerns with that name are alleviated considerably by grouping similar decorators together in a corner of the API that facilitates discovery.

File: shiny/express/_output.py [hunk 1]
```diff
@@ -48,82 +47,11 @@ def wrapper(renderer: RendererBaseT) -> RendererBaseT:
...
-def null_ui(
-    **kwargs: object,
-) -> ui.TagList:
-    return ui.TagList()
-
-
-def null_displayhook(x: object) -> None:
-    pass
+    warn_deprecated(
+        "`express.suspend_display` is deprecated. Please use `express.ui.hide` instead. "
```
Comment (jcheng5) [inline shiny/express/_output.py:54]:
```suggestion
        "`express.suspend_display` is deprecated. Please use `express.ui.hold` instead. "
```

**Comment by jcheng5** [comment]
Just want to state for the record that I'd like this to work (especially if we use the name "hold"):

```python
with hide() as held:
    @render.text()
    def txt2():
        return "Text here"

held
```

**Comment by wch** [comment]
Closing in favor of #1037.

**Pull request closed without merging**